pyof/foundation/base.py

```diff
@@ -558,7 +560,13 @@ def update_header_length(self):
         When sending an OpenFlow message we need to inform the message length
         on the header. This is mandatory.
         """
-        self.header.length = self.get_size()
+        if self.header is not None:
```

Comment:
@macartur I think that here is better to use a try except condition, since we expect all messages  to have a header attribute.

On the new __init_subclass__ I'm proposing this will be more easily handled, since the header existence will be checked during the class creation and, therefore, will probably exist when the update_header_length method get called.

But I am not against doing a check over here. Can you rewrite this part on the commit that added this `if`?

You can force push the change, no problem with that.